Renaming all prometheus metrics to comply with  convention

max-rocket-internet opened the pull request:
Fixes https://github.com/aws/amazon-vpc-cni-k8s/issues/337

The current metric names are not very informative or consistent with [metric and label conventions](https://prometheus.io/docs/practices/naming/). Specifically:

> A metric name should have a (single-word) application prefix relevant to the domain the metric belongs to

So in this PR I am prepending them all with `awscni_`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

micahhausler commented:
Great call. What about the `eni_max` metric? We need to clearly document this in the release notes

max-rocket-internet commented:
Hi @micahhausler 

> What about the `eni_max` metric?

Ahh I missed that one. Fixed. I think that's them all:

```
$ grep -r -A2 '= prometheus.' * | grep Name
ipamd/ipamd.go-			Name: "awscni_ipamd_error_count",
ipamd/ipamd.go-			Name: "awscni_ipamd_action_inprogress",
ipamd/ipamd.go-			Name: "awscni_eni_max",
ipamd/ipamd.go-			Name: "awscni_ip_max",
ipamd/ipamd.go-			Name: "awscni_reconcile_count",
ipamd/ipamd.go-			Name: "awscni_add_ip_req_count",
ipamd/ipamd.go-			Name: "awscni_del_ip_req_count",
ipamd/datastore/data_store.go-			Name: "awscni_eni_allocated",
ipamd/datastore/data_store.go-			Name: "awscni_total_ip_addresses",
ipamd/datastore/data_store.go-			Name: "awscni_assigned_ip_addresses",
pkg/awsutils/awsutils.go-			Name: "awscni_aws_api_lantency_ms",
pkg/awsutils/awsutils.go-			Name: "awscni_aws_api_error_count",
pkg/awsutils/awsutils.go-			Name: "awscni_aws_utils_error_count",
```

> We need to clearly document this in the release notes

True, but that's not done in a PR, right? Or you mean the changelog? OK I've added my change there. Also fixed the formatting a added the missing `1.3.2` release.

micahhausler commented:
@mogren  Also, since this is against master, should the changelog be for 1.4?

mogren commented:
Yes, we squash commits, and yes, it should be against 1.4.0. 

The 1.3.x will be minor patches only, users might be using these metrics so we shouldn't change them in a patch release version.

max-rocket-internet commented:
> it should be against 1.4.0.

OK I've changed my changelog entry to `1.4.0`.

micahhausler commented on docs/troubleshooting.md:
```diff
@@ -150,12 +150,12 @@ ipamd.log.2018-05-16-01  ipamd.log.2018-05-16-06  ipamd.log.2018-05-16-11  ipamd
+# TYPE awscni_eni_allocated gauge
+awscni_eni_allocated 4
 # HELP eni_max The number of maximum ENIs can be attached to the instance
 # TYPE eni_max gauge
```
nit: can you update this name here?

max-rocket-internet replied:
Not a nit! Fixed.

max-rocket-internet replied:
There's more under here, called `go_xxx`. Should I rename those too? I think they come from the prometheus client though.

micahhausler replied:
Correct, those come from the prometheus client and should stay as they are. Thanks!
